core/drivers/firewall/stm32_rifsc.c

```diff
@@ -0,0 +1,563 @@
+	rifsc_drvdata.rif_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG1, regval) != 0;
+	rifsc_drvdata.sec_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG2, regval) != 0;
+	rifsc_drvdata.priv_en = _RIF_FLD_GET(_RIFSC_HWCFGR1_CFG3, regval) != 0;
+
+	regval = io_read32(pdata->base + _RIFSC_HWCFGR2);
+	rifsc_drvdata.nb_risup = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG1, regval);
+	rifsc_drvdata.nb_rimu = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG2, regval);
+	rifsc_drvdata.nb_risal = _RIF_FLD_GET(_RIFSC_HWCFGR2_CFG3, regval);
+
+	pdata->drv_data = &rifsc_drvdata;
+
+	rifsc_drvdata.version = io_read8(pdata->base + _RIFSC_VERR);
+
+	DMSG("RIFSC version %"PRIu32".%"PRIu32,
+	     _RIF_FLD_GET(_RIFSC_VERR_MAJREV, rifsc_drvdata.version),
+	     _RIF_FLD_GET(_RIFSC_VERR_MINREV, rifsc_drvdata.version));
+
+	DMSG("HW cap: enabled[rif:sec:priv]:[%s:%s:%s] nb[risup|rimu|risal]:[%"PRIu8",%"PRIu8",%"PRIu8"]",
+	     rifsc_drvdata.rif_en ? "true" : "false",
+	     rifsc_drvdata.sec_en ? "true" : "false",
+	     rifsc_drvdata.priv_en ? "true" : "false",
+	     rifsc_drvdata.nb_risup,
+	     rifsc_drvdata.nb_rimu,
+	     rifsc_drvdata.nb_risal);
+}
+
+static TEE_Result stm32_rifsc_glock_config(const void *fdt, int node,
+					   struct rifsc_platdata *pdata)
+{
+	const fdt32_t *cuint = NULL;
+	uint32_t glock_conf = 0;
+	int len = 0;
+
+	cuint = fdt_getprop(fdt, node, "st,glocked", &len);
+	if (!cuint) {
+		DMSG("No global lock on RIF configuration");
+		return TEE_SUCCESS;
+	}
+	assert(len == sizeof(uint32_t));
+
+	glock_conf = fdt32_to_cpu(*cuint);
+
+	if (glock_conf & RIFSC_RIMU_GLOCK) {
+		DMSG("Setting global lock on RIMU configuration");
+
+		io_setbits32(pdata->base + _RIFSC_RIMC_CR,
+			     _RIFSC_RIMC_CR_GLOCK);
+
+		if (!(io_read32(pdata->base + _RIFSC_RIMC_CR) &
+		      _RIFSC_RIMC_CR_GLOCK))
+			return TEE_ERROR_ACCESS_DENIED;
+	}
+
+	if (glock_conf & RIFSC_RISUP_GLOCK) {
+		DMSG("Setting global lock on RISUP configuration");
+
+		io_setbits32(pdata->base, _RIFSC_RISC_CR_GLOCK);
+
+		if (!(io_read32(pdata->base) & _RIFSC_RISC_CR_GLOCK))
+			return TEE_ERROR_ACCESS_DENIED;
+	}
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result stm32_rifsc_dt_conf_risup(const void *fdt, int node,
+					    struct rifsc_platdata *pdata)
+{
+	const fdt32_t *conf_list = NULL;
+	unsigned int i = 0;
+	int len = 0;
+
+	conf_list = fdt_getprop(fdt, node, "st,protreg", &len);
+	if (!conf_list) {
+		DMSG("No RISUP configuration in DT");
+		return TEE_ERROR_ITEM_NOT_FOUND;
+	}
+
+	pdata->nrisup = len / sizeof(uint32_t);
+	pdata->risup = calloc(pdata->nrisup, sizeof(*pdata->risup));
+	if (!pdata->risup)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	for (i = 0; i < pdata->nrisup; i++) {
+		uint32_t value = fdt32_to_cpu(conf_list[i]);
+		struct risup_cfg *risup = &pdata->risup[i];
+
+		risup->id = _RIF_FLD_GET(RIFSC_RISC_PER_ID, value);
+		risup->sec = (bool)_RIF_FLD_GET(RIFSC_RISC_SEC, value);
+		risup->priv = (bool)_RIF_FLD_GET(RIFSC_RISC_PRIV, value);
+		risup->lock = (bool)_RIF_FLD_GET(RIFSC_RISC_LOCK, value);
```

Comment:
nitpicking: for consistency with lines 156/158, use the same pattern at those places.